client-internal/src/main/java/org/terracotta/angela/client/Tsa.java

```diff
@@ -374,19 +374,37 @@ public Tsa activateAll() {
 
       if (topology.isNetDisruptionEnabled()) {
         Map<ServerSymbolicName, Integer> proxyTsaPorts = updateToProxiedPorts();
-        int proxyPort = proxyTsaPorts.get(terracottaServer.getServerSymbolicName());
-        IgniteCallable<ConfigToolExecutionResult> callable = () -> {
-          return Agent.controller.configTool(this.instanceId, terracottaServer, cliEnv, "set", "-s", "localhost:" + terracottaServer.getTsaPort(),
-              "-c", "stripe.1.public-hostname=localhost", "-c", "stripe.1.public-port=" + proxyPort);
-        };
-        IgniteClientHelper.executeRemotely(ignite, terracottaServer.getHostname(), ignitePort, callable);
+        for (int i = 0; i < topology.getStripes().size(); ++i) {
+          for (TerracottaServer server : topology.getStripes().get(i)) {
+            int proxyPort = proxyTsaPorts.get(server.getServerSymbolicName());
+            int stripeId = i + 1;
+            int nodeId = getNodeId(server, topology.getStripes().get(i));
+            String publicHostName = "stripe." + stripeId + ".node." + nodeId + ".public-hostname=" + server.getHostName();
+            String publicPort = "stripe." + stripeId + ".node." + nodeId + ".public-port=" + proxyPort;
+            IgniteCallable<ConfigToolExecutionResult> callable = () -> {
+              return Agent.controller.configTool(this.instanceId, terracottaServer, cliEnv,
+                  "set", "-s", server.getHostName() + ":" + server.getTsaPort(),
+                  "-c", publicHostName, "-c", publicPort);
+            };
+            IgniteClientHelper.executeRemotely(ignite, terracottaServer.getHostname(), ignitePort, callable);
+          }
+        }
       }
       return this;
     } else {
       throw new IllegalStateException();
     }
   }
 
+  private int getNodeId(TerracottaServer server, List<TerracottaServer> terracottaServers) {
+    for (int i = 0; i < terracottaServers.size(); ++i) {
+      if (terracottaServers.get(i).equals(server)) {
+        return i + 1;
+      }
+    }
+    throw new IllegalArgumentException("Server:" + server + "should be present in :" + terracottaServers);
```

Comment:
you're missing space, it's going to give a message like:
"Server:Server1should be present in" for a server string "Server1"